spec/ics-23-standard-accumulator/README.md

```diff
@@ -0,0 +1,228 @@
+
+The `calculateRoot` function calculates a succinct commitment to the accumulator state which can be used to verify proofs.
+
+```coffeescript
+calculateRoot(AccumulatorState state) -> AccumulatorRoot
+```
+
+#### Adding & removing elements
+
+The `add` function adds an element to an accumulator.
+
+```coffeescript
+add(AccumulatorState state, Element elem) -> AccumulatorState
+```
+
+The `remove` function removes an element from an accumulator.
+
+```coffeescript
+remove(AccumulatorState state, Element elem) -> AccumulatorState
+```
+
+#### Proof generation
+
+The `createMembershipWitness` function generates a proof that an element has been added to an accumulator.
+
+```coffeescript
+createMembershipWitness(AccumulatorState state, Element elem) -> AccumulatorProof
+```
+
+The `createNonMembershipWitness` function generates a proof that an element has not been added to an accumulator.
+
+```coffeescript
+createNonMembersipWitness(AccumulatorState state, Element elem) -> AccumulatorProof
+```
+
+#### Proof verification
+
+The `verifyMembership` function verifies a proof that an element has been added to an accumulator.
+
+```coffeescript
+verifyMembership(AccumulatorRoot root, AccumulatorProof proof, Element elem) -> boolean
+```
+
+The `verifyNonMembership` function verifies a proof that an element has not been added to an accumulator.
+
+```coffeescript
+verifyNonMembership(AccumulatorRoot root, AccumulatorProof proof, Element elem) -> boolean
+```
+
+### Optional functions
+
+An accumulator construction MAY provide the following functions:
+
+The `batchVerifyMembership` function verifies a proof that many elements have been added to an accumulator.
+
+```coffeescript
+batchVerifyMembership(AccumulatorRoot root, AccumulatorProof proof, Set<Element> elems) -> boolean
+```
+
+The `batchVerifyNonMembership` function verifies a proof that many elements have not been added to an accumulator.
+
+```coffeescript
+batchVerifyNonMembership(AccumulatorRoot root, AccumulatorProof proof, Set<Element> elems) -> boolean
+```
+
+If defined, these functions MUST be computationally equivalent to the conjunctive union of `verifyMembership` and `verifyNonMembership` respectively (`proof` may vary):
+
+```coffeescript
+batchVerifyMembership(root, proof, elems) ==
+  verifyMembership(root, proof, elems[0]) &&
+  verifyMembership(root, proof, elems[1]) && ...
+```
+
+```coffeescript
+batchVerifyMembership(root, proof, elems) ==
+  verifyNonMembership(root, proof, elems[0]) &&
+  verifyNonMembership(root, proof, elems[1]) && ...
+```
+
+If batch verification is possible and more efficient than individual verification of one proof per element, an accumulator construction SHOULD define batch verification functions.
+
+### Properties
+
+Accumulators must be *correct* and *sound*. In practice, violations of these properties by computationally-bounded adversaries may be negligible in some security parameter, which is sufficient for use in IBC.
+
+#### Correctness
```

Comment:
This is typically defined as Completeness

Reply:
Oh really? https://eprint.iacr.org/2017/043.pdf uses "correctness"

I like completeness better though

Reply:
oh interesting. I imagine it varies per author then, completeness is definitely the one I've seen in more locations.

Reply:
OK, agreed, updated.